zenodo/modules/deposit/api.py

```diff
@@ -213,3 +218,39 @@ def _publish_edited(self):
         if not record['communities']:
             del record['communities']
         return record
+
+    @staticmethod
+    def _build_agent_info():
+        """Build the SIP agent info."""
+        email = ''
+        ip_address = ''
```

Comment:
I guess this will not happen, but we can still log the IP address even for non-logged in user. If you are thinking about programatic API usage without a request, then you can also check with ``has_request_context()`` from flask.